### PR TITLE
CHORE:  a better gitignore  file for ignore  no extension binary files correctly

### PR DESCRIPTION
**Pull request by codeudan**
Currently, gitignore file is kind of redundant for ignore binary files and some Clang build stuff.

This is a better ignore file to ignore no extension binary files in src dir 😄

### CODE REVIEW

**Review by Aemiii91 — APPROVED** [review]
Neat :)

**Review by Aemiii91 — APPROVED** [review]

**Comment by codeudan** [comment]
Fixed some problems.  Now can merge it.

**Comment by Aemiii91** [comment]
How does this handle `Makefile` ?

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,15 @@
+# only ignore all extension less binary file under src
+src/**/**
+!src/**/
+!src/*.*
```
Comment (Aemiii91) [inline .gitignore:4]: Please add `!Makefile` and `!easter4`.

File: .gitignore [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
+
+# put other need ignore
+src/**/*.o
+src/**/*.exe
```
Comment (Aemiii91) [inline .gitignore:8]: Why would we have `.exe` files?

**Review by Aemiii91 — CHANGES_REQUESTED** [review]

**Comment by Aemiii91** [comment]
Closed as inactive

**Pull request closed without merging**